src/app/components/pages/admin-cell/admin-cell.component.ts

```diff
@@ -315,10 +343,147 @@ export class AdminCellComponent implements OnInit {
           }
         }
       );
+  }
+
+  updateEvent(eventId: number) {
+    this.eventService.updateEvent(eventId, this.eventForm.value).subscribe(
+        data => {
+          this.myModalService.get('event modal').toggle();
+          this.resetForm();
+          this.notificationService.success('Modification réussie',
+            `La plage horaire a été modifiée`);
+
+          this.get_events();
+        },
+        err => {
+
+          if (err.error.nb_volunteers_needed) {
+            this.eventForm.controls['nb_volunteers_needed'].setErrors({
+              apiError: err.error.nb_volunteers_needed
+            });
+          }
+          if (err.error.cycle_id) {
+            this.eventForm.controls['cycle_id'].setErrors({
+              apiError: err.error.cycle_id
+            });
+          }
+          if (err.error.task_type_id) {
+            this.eventForm.controls['task_type_id'].setErrors({
+              apiError: err.error.task_type_id
+            });
+          }
+          if (err.error.start_date) {
+            this.eventForm.controls['start_date'].setErrors({
+              apiError: err.error.start_date
+            });
+          }
+          if (err.error.end_date) {
+            this.eventForm.controls['end_date'].setErrors({
+              apiError: err.error.end_date
+            });
+          }
+          if (err.error.non_field_errors) {
+            this.eventForm.setErrors({
+              apiError: err.error.non_field_errors
+            });
+          }
+        }
+      );
+  }
+
+  opEvent() {
+
+    if (this.eventForm.valid) {
+      if (this.modalCreate) {
+        this.createEvent();
+      } else {
+        this.updateEvent(this.modalEventId);
+      }
+
     } else {
       for (const controlKey of Object.keys(this.eventForm.controls)) {
         this.eventForm.controls[controlKey].markAsTouched();
       }
     }
   }
+
+  getVolunteersField(event: Event) {
+    return event.nb_volunteers + ' / ' + event.nb_volunteers_needed;
+  }
+  getStandByField(event: Event) {
+    return event.nb_volunteers_standby + ' / ' + event.nb_volunteers_standby_needed;
+  }
```

Comment:
These two functions can have a better places in event model so we can use them anywhere.
I'm sure we will use this kind of display a lot in all the platform.

Reply:
Good point ! Moved !